libs/partners/databricks/langchain_databricks/utils.py

```diff
@@ -0,0 +1,101 @@
+from typing import Any, List, Union
+from urllib.parse import urlparse
+
+import numpy as np
+
+
+def get_deployment_client(target_uri: str) -> Any:
+    if (target_uri != "databricks") and (urlparse(target_uri).scheme != "databricks"):
+        raise ValueError(
+            "Invalid target URI. The target URI must be a valid databricks URI."
+        )
+
+    try:
+        from mlflow.deployments import get_deploy_client  # type: ignore[import-untyped]
+
+        return get_deploy_client(target_uri)
+    except ImportError as e:
+        raise ImportError(
+            "Failed to create the client. "
+            "Please run `pip install mlflow` to install "
+            "required dependencies."
+        ) from e
+
+
+# Utility function for Maximal Marginal Relevance (MMR) reranking.
+# Copied from langchain_community/vectorstores/utils.py to avoid cross-dependency
+Matrix = Union[List[List[float]], List[np.ndarray], np.ndarray]
+
+
+def maximal_marginal_relevance(
```

Comment:
@efriis This MMR utility is used in most of partner vector store implementations, but the code is duplicated to each package for avoiding dependency to the community package. What about moving this into the core package?
https://github.com/search?q=repo%3Alangchain-ai%2Flangchain%20maximal_marginal_relevance&type=code

Reply:
#25689 merged!